packages/synapse-client/src/SynapseClient.ts

```diff
@@ -41,17 +41,56 @@ import { VerificationServicesApi } from './generated/apis/VerificationServicesAp
 import { WebhookServicesApi } from './generated/apis/WebhookServicesApi'
 import { WikiPageServices2Api } from './generated/apis/WikiPageServices2Api'
 import { WikiPageServicesApi } from './generated/apis/WikiPageServicesApi'
-import { BaseAPI, Configuration } from './generated/runtime'
-import { fetchWithExponentialTimeout } from './util/fetchWithExponentialTimeout'
+import { ErrorResponse } from './generated/models/ErrorResponse'
+import {
+  Configuration,
+  ConfigurationParameters,
+  ErrorContext,
+} from './generated/runtime'
+import { fetchResponseWithExponentialTimeout } from './util/fetchWithExponentialTimeout'
+import { SynapseClientError } from './util/SynapseClientError'
 
-const DEFAULT_CONFIG = new Configuration({
-  fetchApi: fetchWithExponentialTimeout,
-})
+const DEFAULT_CONFIG_PARAMETERS: ConfigurationParameters = {
+  fetchApi: fetchResponseWithExponentialTimeout,
+  middleware: [
```

Comment:
The generated client throws its own error class, which breaks all of our existing error handling code. This middleware will inspect the error and throw (or rethrow) a SynapseClientError when appropriate.